tests/integration/suite/actors/reminders/scheduler/metrics.go

```diff
@@ -0,0 +1,210 @@
+		Endpoints:   []string{fmt.Sprintf("localhost:%d", m.etcdPort)},
+		DialTimeout: 5 * time.Second,
+	})
+
+	// Use "path/filepath" import, it is using OS specific path separator unlike "path"
+	etcdKeysPrefix := filepath.Join("dapr", "jobs")
+
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		keys, rerr := etcdClient.ListAllKeys(ctx, etcdKeysPrefix)
+		require.NoError(c, rerr)
+		assert.Empty(c, keys)
+	}, time.Second*10, 10*time.Millisecond)
+
+	// assert false, since count is 0 here
+	m.assertMetricExists(t, ctx, frameworkClient, "dapr_scheduler_jobs_created_total", 0)
+
+	_, err := client.InvokeActor(ctx, &runtimev1pb.InvokeActorRequest{
+		ActorType: "myactortype",
+		ActorId:   "myactorid",
+		Method:    "foo",
+	})
+	require.NoError(t, err)
+
+	_, err = client.RegisterActorReminder(ctx, &runtimev1pb.RegisterActorReminderRequest{
+		ActorType: "myactortype",
+		ActorId:   "myactorid",
+		Name:      "remindermethod",
+		DueTime:   "0s",
+		Period:    "1s",
+	})
+	require.NoError(t, err)
+
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		keys, rerr := etcdClient.ListAllKeys(ctx, etcdKeysPrefix)
+		require.NoError(c, rerr)
+		assert.Len(c, keys, 1)
+	}, time.Second*10, 10*time.Millisecond)
+
+	// true, since count is 1
+	m.assertMetricExists(t, ctx, frameworkClient, "dapr_scheduler_jobs_created_total", 1)
+
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		assert.GreaterOrEqual(c, m.triggered.Load(), int64(1))
+	}, 30*time.Second, 10*time.Millisecond, fmt.Sprintf("failed to wait for 'triggered' to be greatrer or equal 1, actual value %d", m.triggered.Load()))
+}
+
+// assert the metric exists and the count is correct
+func (m *metrics) assertMetricExists(t *testing.T, ctx context.Context, client *http.Client, expectedMetric string, expectedCount int) {
+	t.Helper()
+
+	metricReq, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("http://localhost:%d/metrics", m.scheduler.MetricsPort()), nil)
+	require.NoError(t, err)
+
+	resp, err := client.Do(metricReq)
+	require.NoError(t, err)
+
+	respBody, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.NoError(t, resp.Body.Close())
+
+	foundMetric := false
+
+	for _, line := range bytes.Split(respBody, []byte("\n")) {
+		if len(line) == 0 || line[0] == '#' {
+			continue
+		}
+
+		split := bytes.Split(line, []byte(" "))
+		if len(split) != 2 {
+			continue
+		}
+
+		// dapr_scheduler_jobs_created_total{app_id="appid"}
+		metricName := string(split[0])
+		metricVal := string(split[1])
+		if !strings.Contains(metricName, expectedMetric) {
+			continue
+		}
+		if strings.Contains(metricName, expectedMetric) {
+			metricCount, err := strconv.Atoi(metricVal)
+			require.NoError(t, err)
+			assert.Equal(t, expectedCount, metricCount)
+			foundMetric = true
+			break
+		}
+	}
+	if expectedCount > 0 {
+		assert.True(t, foundMetric, "Expected metric %s not found", expectedMetric)
+	} else {
+		assert.False(t, foundMetric)
+	}
+}
```

Comment:
We should move the metrics call to the Scheduler process `func (s *Scheduler) Metrics(t *testing.T, ctx context.Context) map[string]float64`.

This is what we are doing for daprd https://github.com/dapr/dapr/blob/b26a1951c482b6754d7486c0a1ec0b9999eb99a0/tests/integration/framework/process/daprd/daprd.go#L328

Reply:
that really cleans things up. Wish I saw this earlier!